main.js

```diff
@@ -0,0 +1,107 @@
+const gameBoard = (() => {
+  const initBoard = () => document.querySelectorAll('.gameboard').forEach(element => {
+      element.innerText = '';
+      element.classList.remove('clicked');
+  });
+  return {
+      initBoard
+  };
+})();
+
+const game = (() => {
+  const winCombo = [
+  ['1', '2', '3'], 
+  ['4', '5', '6'], 
+  ['7', '8', '9'], 
+  ['1', '4', '7'], 
+  ['2', '5', '8'], 
+  ['3', '6', '9'], 
+  ['1', '5', '9'], 
+  ['3', '5', '7']
+];
+  let stepCount = 0;
+  const getStepCount = () => stepCount;
+  const recordGame = () => stepCount++;
+  const checkGame = (player) => {
+      return winCombo.some(combo => combo.every(element => player.getPlayerStep().includes(element)));
+  }
+  const resetGame = () => stepCount = 0;
+  return {
+      recordGame,
+      getStepCount,
+      checkGame,
+      resetGame
+  };
+})();
+
+const Player = (marker) => {
+  let playerStep = [];
+  const getPlayerStep = () => playerStep;
+  const addStep = (position) => playerStep.push(position);
+  const resetPlayer = () => playerStep = [];
+  return {
+      addStep,
+      getPlayerStep,
+      resetPlayer,
+      marker
+  };
+};
+
+const player1 = Player('x');
```

Comment:
You should keep all your global variable definitions together. Best practices are to put them at the top of the file so that they will always be defined before you try to use them.